### PR TITLE
Fix notification replies for iOS

### PR DESCRIPTION
**Pull request by csduarte**
This PR fixes an issue where replies to iOS notifications would not complete if the app was shutdown

### CODE REVIEW

File: app/mattermost.js [hunk 1]
```diff
@@ -521,17 +522,26 @@ export default class Mattermost {
...
-        }
+                if (badge >= 0) {
+                    PushNotifications.setApplicationIconBadgeNumber(badge);
+                }
 
-        if (completed) {
-            // You must call to completed(), otherwise the action will not be triggered
-            completed();
+                this.replyNotificationData = null;
+            }).then(() => {
```
Comment (hmhealey) [inline app/mattermost.js]: I think this can just be 
```
}).then(completed);
```

**Comment by jarredwitt** [comment]
@hmhealey refactored

File: app/mattermost.js [hunk 2]
```diff
@@ -496,7 +497,7 @@ export default class Mattermost {
         }
     };
 
-    onPushNotificationReply = (data, text, badge, completed) => {
+    onPushNotificationReply = (data, text, badge, completed = emptyFunction) => {
```
Comment (hmhealey) [inline app/mattermost.js]: We don't actually need the default value here. `new Promise(func).then(undefined);` doesn't throw an error

**Comment by jarredwitt** [comment]
@hmhealey ready for you again.

**Pull request merged**